test/e2e/volume_provisioning.go

```diff
@@ -109,16 +109,16 @@ var _ = KubeDescribe("Dynamic provisioning", func() {
 			// 10 minutes here. There is no way how to see if kubelet is
 			// finished with cleaning volumes. A small sleep here actually
 			// speeds up the test!
-			// One minute should be enough to clean up the pods properly.
-			// Detaching e.g. a Cinder volume takes some time.
+			// Three minutes should be enough to clean up the pods properly.
+			// We've seen GCE PD detach to take more than 1 minute.
 			By("Sleeping to let kubelet destroy all pods")
-			time.Sleep(time.Minute)
+			time.Sleep(3 * time.Minute)
```

Comment:
No, it is not. As you can see few lines above, the test *does* wait for pod termination. And as described in the comment, when a pod is deleted all its volumes are still attached to the node and kubelet (slowly) starts detaching them. There is no event that would signalize that kubelet finished and all volumes are detached.